include/boost/geometry/algorithms/densify.hpp

```diff
@@ -92,11 +89,11 @@ struct densify_range
         auto strategy = strategies.densify(rng);
         push_back_policy<MutRng> policy(rng_out);
 
-        iterator_t prev = it;
+        auto prev = it;
         for ( ++it ; it != end ; prev = it++)
         {
-            point_t const& p0 = *prev;
-            point_t const& p1 = *it;
+            auto const& p0 = *prev;
+            auto const& p1 = *it;
```

Comment:
Note that here iterator could return non-true reference type. Previously this type was converted to `point_t` which was value_type vs reference_type returned by interator. If this non-true reference type was not adapted to Point concept the call to convert below could fail to compile. I don't know if that's a problem.

Reply:
I'm not sure either. Reverted (it was not related to the iterator anyway)
:heavy_check_mark: